"""
>>> import sys
>>> sys.path.append('..')
>>> lazy_matrix_mul = __import__('101-lazy_matrix_mul').lazy_matrix_mul
>>> lazy_matrix_mul([[1, 2], [3, 4]], [[1, 2], [3, 4]])
array([[ 7, 10],
       [15, 22]])
>>> lazy_matrix_mul([[1, 2]], [[3, 4], [5, 6]])
array([[13, 16]])
>>> lazy_matrix_mul([[1, 2], [3, 4]], [[1, 2], [3, 4]])
array([[ 7, 10],
       [15, 22]])
>>> lazy_matrix_mul([[1, 2]], [[3, 4], [5, 6]])
array([[13, 16]])
>>> lazy_matrix_mul([], [[1]])
Traceback (most recent call last):
...
IndexError: tuple index out of range
>>> lazy_matrix_mul([[1,2],[3,4]], [[]])
Traceback (most recent call last):
...
ValueError: shapes (2,2) and (1,0) not aligned: 2 (dim 1) != 1 (dim 0)
>>> lazy_matrix_mul('x', [[1,2,4]])
Traceback (most recent call last):
...
TypeError: Scalar operands are not allowed, use '*' instead
>>> lazy_matrix_mul([[1,2],[3,2]], [1,'x',2,3])
Traceback (most recent call last):
...
TypeError: 'int' object is not iterable
>>> lazy_matrix_mul([[1,2,3]],[[1,2,'x']])
Traceback (most recent call last):
...
ValueError: invalid data type for einsum
>>> lazy_matrix_mul([[1,2,3], [1,2]], [[1]])
Traceback (most recent call last):
...
Exception: setting an array element with a sequence.
>>> lazy_matrix_mul([[1,2,3]], [[1,2,3,4]])
Traceback (most recent call last):
...
ValueError: shapes (1,3) and (1,4) not aligned: 3 (dim 1) != 1 (dim 0)
>>> lazy_matrix_mul()
Traceback (most recent call last):
...
TypeError: lazy_matrix_mul() missing 2 required positional arguments: 'm_a' and 'm_b'
>>> lazy_matrix_mul([[1,2]])
Traceback (most recent call last):
...
TypeError: lazy_matrix_mul() missing 1 required positional argument: 'm_b'
>>> lazy_matrix_mul([[1,2]], 'x')
Traceback (most recent call last):
...
TypeError: Scalar operands are not allowed, use '*' instead
>>> lazy_matrix_mul([1,2,3], [[1,2], [2,3]])
Traceback (most recent call last):
...
TypeError: 'int' object is not iterable
>>> lazy_matrix_mul([['1', '2'], ['3', '4']], [[1,3]])
Traceback (most recent call last):
...
ValueError: invalid data type for einsum
>>> lazy_matrix_mul([[1,2],[3,4]], [[1,2], [2]])
Traceback (most recent call last):
...
Exception: setting an array element with a sequence.

"""
